flight/PiOS/posix/pios_delay.c

```diff
@@ -166,3 +166,21 @@ uint32_t PIOS_DELAY_DiffuS2(uint32_t raw, uint32_t later) {
 	return diff;
 }
 
+/**
+ * @brief Query the Delay timer for the current uS 
+ * @return A microsecond value
+ */
+uint32_t PIOS_DELAY_GetuS()
+{
+	return PIOS_DELAY_GetRaw();
+}
+
+/**
+ * @brief Calculate time in microseconds since a previous time
+ * @param[in] t previous time
+ * @return time in us since previous time t.
+ */
+uint32_t PIOS_DELAY_GetuSSince(uint32_t t)
+{
+	return PIOS_DELAY_GetuS() - t;
```

Comment:
Doesn't handle overflow.

Reply:
Isn't this the best you can do and still conform to the existing API?

Reply:
Ah, unsigned arithmetic does handle it correctly already.. carry on. 👍